crates/typos-dict/assets/words.csv

```diff
@@ -32948,6 +33022,7 @@ nigthmares,nightmares
 nihilim,nihilism
 nihilisim,nihilism
 nihilsim,nihilism
+nilable,nillable
```

Comment:
Not sure the best way to go about this, but `nilable` is actually fairly common in the Ruby world. E.g. https://sorbet.org/docs/nilable-types.

I worked around it by just adding it to `extended-words`, which is fine if we think this is the preferred spelling outside of certain ecosystems, but it could be a point of false positives.

Reply:
I'd rather allow more typos than correct things that shouldn't be typos.

@Blacksmoke16 would it be ok for `nilable` to be in the default `extend-words` for Ruby only or would it appear in enough other contexts to need to be generally allowed?

Reply:
> but am also curious about how best to handle common spelling mistakes that are widely accepted in certain languages/frameworks/communities

IMO it would be nice if there was a more selective way to denote a typo is expected. E.g. such that I could mark a specific usage of `referer` as correct, but not prevent that from flagging other contexts. Currently, afaik, the only way to handle that is to just globally accept the typo by adding it to `extended-words`, or accept it for a given "type" of file.

> would it be ok for `nilable` to be in the default `extend-words` for Ruby only

For my particular use case, it would need to be Ruby (.rb), Crystal (.cr), and Markdown (.md) in order to cover usages and documentation related to the code. Did a quick google of `"nilable"` and saw some things in there about some Go libraries, and ClojureScript. 
 So guess it depends on how many contexts is enough to suggest it should be globally allowed.

Reply:
Created #635 for nilable.

For referer, we have a couple of issues around narrowing things.  If one doesn't fit, feel free to create a new issue.